### DISCUSSION TITLE
VSCode Copilot "Sign in to Github" fails trying to save "did-authenticate?nonce..." file.

### DISCUSSION THREAD

**Discussion by HendrikJan** [community/community#15314] [Copilot Conversations] [answered]
VSCode Copilot did work a wile ago, but now fails to authenticate with our organisation.

I get a popup to sign in to Github:
![image](https://user-images.githubusercontent.com/2218920/164429650-e73b9cd6-7945-4eea-9360-fe06f34f4b77.png)

Then a question to permit Copilot:
![image](https://user-images.githubusercontent.com/2218920/164429886-ce70fb3f-b4cd-4471-9669-1af6c82ef4e4.png)

Then a browser window asking to authenticate with our organisation.
After authenticating with our organisation, VSCode opens a file with a very long name:
![image](https://user-images.githubusercontent.com/2218920/164430175-96dbb212-b44f-41fb-8e07-416713024478.png)

When I try to save this file, I get an error:
![image](https://user-images.githubusercontent.com/2218920/164430292-26a619a8-3236-4073-ae93-5afc1368d6fc.png)

The error also contains this text: 
```
Failed to save 'did-authenticate?nonce... (Unknown (FileSystemError): Error: ENAMETOOLONG: name too long, open '...')
```

If I close the file without saving, then Copilot is not authenticated and will ask again the next time I start VSCode.
I cannot save the file, so I cannot authenticate Copilot.

My operating system is Ubuntu.

**Comment by davecheney** [comment]
Can you please check that you’re using VScode, not VScodiun or OSSCode. The latter are often packaged by linux distributions and do not currently work with Copiot.

**Reply by HendrikJan** [reply]
Version: 1.66.2
Commit: dfd34e8260c270da74b5c2d86d61aee4b6d56977
Date: 2022-04-11T07:49:24.808Z
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Linux x64 5.13.0-39-generic snap

**Reply by HendrikJan** [reply]
![image](https://user-images.githubusercontent.com/2218920/164458826-c2171452-f336-488e-8679-84cfbc452ea0.png)

**Reply by davecheney** [reply]
I think if you use the vscode from https://code.visualstudio.com/download copilot will work.

**Comment by HendrikJan** [comment, marked as answer]
As @davecheney said: I had to re-install VSCode from https://code.visualstudio.com/download
